src/sys/resource.rs

```diff
@@ -0,0 +1,48 @@
+use std::mem;
+
+use libc::{self, c_int};
+
+use {Errno, Result};
+
+#[repr(i32)]
+pub enum Resource {
```

Comment:
`Resource` as a name still feels a little off give the values in the enum.  You are not really getting the rlimit of a resource, you are getting the value of a specific resource limit in the system (one "resource" may have multiple related limits).

I'm not completely sold on this, but I could see naming the enum `ResourceLimit` and allowing get/set on the limit.  E.g.

```rust
let limit =  ResourceLimit::RLIMIT_STACK.get().unwrap();
ResourceLimit::RLIMIT_STACK.set(limit).unwrap();
```